data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,441 @@
+            job_results = bq_utils.query(q=query_dict['query'], batch=True)
+            logging.info('%s table written to %s' %
+                         (query_dict['destination_table_id'],
+                          query_dict['destination_dataset_id']))
+            query_job_id = job_results['jobReference']['jobId']
+            query_job_ids.append(query_job_id)
+        else:
+            logging.info('Truncating table with clean data, using query %s' %
+                         (query_dict['query']))
+            job_results = bq_utils.query(q=query_dict['query'], batch=True)
+            logging.info('%s table updated with clean rows in %s' %
+                         (query_dict['destination_table_id'],
+                          query_dict['destination_dataset_id']))
+            query_job_id = job_results['jobReference']['jobId']
+            query_job_ids.append(query_job_id)
+
+    incomplete_jobs = bq_utils.wait_on_jobs(query_job_ids)
+    if incomplete_jobs:
+        logging.info('Failed on {count} job ids {ids}'.format(
+            count=len(incomplete_jobs), ids=incomplete_jobs))
+        logging.info('Terminating retraction')
+        raise bq_utils.BigQueryJobWaitError(incomplete_jobs)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=
+        'Runs retraction of deactivated EHR participants on all datasets in project. The rows to be retracted'
+        'are based on the date entered and will be retracted if the date entered is before the retraction date'
+        'Uses project_id, deactivated pid_table_id to determine '
+        'the pids to retract data for. The pid_table_id needs to contain '
+        'the person_id and research_id and deactivated_date columns specified in the schema above, '
+        'but research_id can be null if deid has not been run yet. ',
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-p',
+                        '--project_id',
+                        action='store',
+                        dest='project_id',
+                        help='Identifies the project to retract data from',
+                        required=True)
+    parser.add_argument('-t',
+                        '--ticket_number',
+                        action='store',
+                        dest='ticket_number',
+                        help='Ticket number to append to sandbox table names',
+                        required=True)
+    parser.add_argument(
+        '-pp',
+        '--pids_project_id',
+        action='store',
+        dest='pids_project_id',
+        help='Identifies the project where the pids table is stored',
+        required=True)
+    parser.add_argument(
+        '-pd',
+        '--pids_dataset_id',
+        action='store',
+        dest='pids_dataset_id',
+        help='Identifies the dataset where the pids table is stored',
+        required=True)
+    parser.add_argument('-pt',
```

Comment:
This is a little odd.  The short form of a command is typically `-` followed by a single alphabetic character.  This allows chaining multiple single character short form commands into a single long chain, for example `-s`, `-t`, `-u` could be invoked from the command line as `-stu`.  I tested your argument parsing code and it works, but it isn't entirely conventional.  When I think about conventional short form options I think about common linux/macOS command options.  You can view examples of what I consider common command line conventions by using `man ls`, `man cat`, or `man tail`.  Use `q` to exit a man page.

This link references the common short form and long form of command line options.
https://docs.python.org/3/library/argparse.html#name-or-flags

One other thing is I might not mix the underscore and hyphen in the command line parameters.  We haven't set a precedent for this in our project yet, but I think `--pids-project-id` is a little cleaner than mixing them like `--pids_project_id`.

Reply:
thanks for the reference link!